File: src/models/UserFollow.ts
import {Entity,ManyToOne,JoinColumn,CreateDateColumn,PrimaryColumn} from 'typeorm';
import { User } from './User';

@Entity('user_follow')
export class UserFollow {
  @PrimaryColumn()
  user_id!: number;

  @PrimaryColumn()
  following_user_id!: number;

  @CreateDateColumn()
  followed_at!: Date;

  @ManyToOne(() => User, (user) => user.following)
  @JoinColumn({ name: 'user_id' })
  user!: User;

  @ManyToOne(() => User, (user) => user.followers)
  @JoinColumn({ name: 'following_user_id' })
  followingUser!: User;
}

File: src/models/user_follow.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from src.db.database import Base


class UserFollow(Base):
    __tablename__ = "user_follow"

    user_id = Column(Integer, ForeignKey("users.id"), primary_key=True)
    following_user_id = Column(Integer, ForeignKey("users.id"), primary_key=True)
    followed_at = Column(DateTime(timezone=True), server_default=func.now())

    user = relationship("User", foreign_keys=[user_id], back_populates="following")
    following_user = relationship(
        "User", foreign_keys=[following_user_id], back_populates="followers"
    )
